api/service.go

```diff
@@ -0,0 +1,408 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/decred/dcrd/chaincfg/v3"
+	chainjson "github.com/decred/dcrd/rpc/jsonrpc/types/v3"
+	apiTypes "github.com/decred/dcrdata/v7/api/types"
+)
+
+type Service struct {
+	client      *Client
+	chainParams *chaincfg.Params
+}
+
+type Exchange int
+
+const (
+	Bittrex Exchange = iota
+	Binance
+	KuCoin
+)
+
+const (
+	mainnetBaseUrl        = "https://mainnet.dcrdata.org/"
+	testnetBaseUrl        = "https://testnet.dcrdata.org/"
+	blockbookMainnet      = "https://blockbook.decred.org:9161/"
+	blockbookTestnet      = "https://blockbook.decred.org:19161/"
+	binanceBaseUrl        = "https://api.binance.com"
+	binanceTestnetBaseUrl = "https://testnet.binance.vision"
+	bittrexBaseUrl        = "https://api.bittrex.com/v3"
+	kucoinBaseUrl         = "https://api.kucoin.com"
+	KuCoinTestnetBaseUrl  = "https://openapi-sandbox.kucoin.com"
+)
+
+var (
+	supportedExchanges = []string{"Bittrex", "Binance", "KuCoin"}
+)
+
+func NewService(chainParams *chaincfg.Params) *Service {
+	conf := &ClientConf{
+		Debug: true,
+	}
+	if chainParams.Name == chaincfg.TestNet3Params().Name {
+		conf.BaseUrl = testnetBaseUrl
+	} else {
+		conf.BaseUrl = mainnetBaseUrl
+	}
+
+	client := NewClient(conf)
+	client.ReqFilter = func(info RequestInfo) (req *http.Request, err error) {
+
+		req, err = http.NewRequest(info.Method, info.Url, bytes.NewBuffer([]byte(info.Payload.(string))))
+		if err != nil {
+			log.Error(err)
+			return
```

Comment:
probably better log errors once, at the very top instead of here (to keep it consistent)